src/containers/Application/Part0.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react'
+import { useLocation } from 'wouter'
+import VaccineInfo from '../../components/ApplicationForm/VaccineInfo'
+import NavigationButtons from '../../components/NavigationButtons'
+import VerticalProgressBar from '../../components/VerticalProgressBar'
+import { useAuth } from '../../utility/Auth'
+import { useHackerApplication } from '../../utility/HackerApplicationContext'
+import { checkForError, validateFormSection } from '../../utility/Validation'
+
+export default () => {
+  const { application, updateApplication, forceSave } = useHackerApplication()
+  const [, setLocation] = useLocation()
+  const { logout } = useAuth()
+  const [errors, setErrors] = useState({})
+  const [loading, setLoading] = useState(false)
+
+  const validate = change => {
+    const newErrors = validateFormSection(change, 'vaccineInfo')
+    setErrors({ ...errors, ...newErrors })
+    return { ...errors, ...newErrors }
+  }
+
+  const save = async () => {
+    setLoading(true)
+    await forceSave()
+    setLoading(false)
+  }
+
+  const updateVaccineInfo = change => {
+    validate(change)
+    console.log(application, 'this is the application object')
+
+    updateApplication({
+      vaccineInfo: {
+        ...change,
+      },
+    })
+  }
+
+  /**
+   * Saves and moves to next page
+   */
+  const handleNavigation = async href => {
+    await save()
+    if (href === '/application/part-1') {
+      const newErrors = validate(application.vaccineInfo)
+      if (checkForError(newErrors)) {
+        window.alert('You can only participate in nwHacks if you are double-vaccinated by then!')
+        return
+      }
```

Comment:
```suggestion
```
If you wanted to show a custom message for the validation function, you can add a new message constant in `validation.js` and use it as the validator.

Reply:
Got it! This also removes the ugly window alert and replaces it by an error message instead!